mm2src/coins/utxo.rs

```diff
@@ -2649,3 +2649,69 @@ fn kmd_interest(height: Option<u64>, value: u64, lock_time: u64, current_time: u
     minutes -= 59;
     (value / 10_512_000) * minutes
 }
+
+fn kmd_interest_accrue_stop_at(height: u64, lock_time: u64) -> u64 {
+    let minutes = if height < 1_000_000 {
+        // interest stop accruing after 1 year before block 1000000
+        365 * 24 * 60
+    } else {
+        // interest stop accruing after 1 month past 1000000 block
+        31 * 24 * 60
+    };
+
+    lock_time + minutes
+}
+
+#[derive(Deserialize, Eq, PartialEq, Serialize)]
+pub struct KmdRewardInfoElement {
+    tx_hash: H256Json,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    height: Option<u64>,
+    amount: u64,
```

Comment:
Please also change the amount and accrued_rewards to BigDecimal so these fields will display the denominated amount, not satoshis.